dashboard/src/components/Catalog/CatalogItem.tsx

```diff
@@ -37,18 +38,19 @@ function trimDescription(desc: string): string {
 
 const CatalogItem: React.SFC<ICatalogItemProps> = props => {
   const { item } = props;
-  const { icon, name, repoName, version, description, type, namespace, id, csv } = item;
+  const { icon, name, repo, version, description, type, namespace, id, csv } = item;
   const iconSrc = icon || placeholder;
   let link;
   let tag1;
   let subIcon;
```

Comment:
if the repo is mandatory and it's needed to render this card, you can add a check `if (!repo) return null` or smth like that to avoid all the `repo?.` that may indicate that the repo is not mandatory

Reply:
> if the repo is mandatory and it's needed to render this card, you can add a check `if (!repo) return null` or smth like that

That would be swallowing an error in a different way, imo? We need to be rendering something - a chart or an operator. If it's a chart, then the repo is required. Currently we would swallow the error by rendering a chart with invalid links etc. If this was the way to fix it, I'd instead suggest throwing an error if it is a chart type and repo is null, but...

> to avoid all the `repo?.` that may indicate that the repo is not mandatory

So the `repo` is necessary if the catalog item is a chart. I think the issue might be the data modelling here: we should be specifying requirements for a chart catalog item separately from an operator catalog item (though they may share many properties), so that this type requirement (that a repo is required for a chart) can be explicit. Thoughts?